tests/testthat/test-build-articles.R

```diff
@@ -4,3 +4,40 @@ test_that("can recognise intro variants", {
   expect_true(article_is_intro("pack-age", "pack.age"))
   expect_true(article_is_intro("articles/pack-age", "pack.age"))
 })
+
+test_that("can build article that uses html_vignette", {
+  pkg <- as_pkgdown(test_path("assets/article-as-is"))
+  withr::defer(clean_site(pkg))
```

Comment:
This could become a helper for all tests? Maybe even encompassing the two lines, `local_pkgdown_website(path)`.